### PR TITLE
EZP-24961: Combine template modules

### PR DESCRIPTION
**Pull request by michalpipa**
> Jira issue: https://jira.ez.no/browse/EZP-24961
> Status: ready for review
> Dependencies: #427

Enables combining template modules in combo loader.

### CODE REVIEW

File: Loader/ComboLoader.php [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+    public function combineFilesContent(array $files)
+    {
+        $content = '';
+
+        foreach ($files as $file) {
+            $ext = $this->findFileExtension($file);
+
+            if ($ext === Loader::EXT_JS) {
+                if ($this->isTemplateFile($file)) {
```
Comment (bdunogier) [inline Loader/ComboLoader.php]: The `if` is getting a bit too big imgo.

I'd abstract the type test (`$ext === Loader::EXT_JS`) into meaningful methods:
```php
if ($this->isTemplateFile($file)) {
    $content .= $this->renderTemplate($file);
} elseif ($this->isYuiFile($file) || $this->isCssFile($file)) {
    $file = $this->fixUnresolvedPath($file);
    $content .= $this->loadFile($file, $ext);
} else {
    throw new NotFoundException('file type', $file);
}
```

**Comment by bdunogier** [comment]
We would need / have needed this for the RC1 tomorrow, @michalpipa. Any possibility ? We can fix bugs after that.

**Comment by michalpipa** [comment]
@bdunogier: rebased and ready

**Comment by michalpipa** [comment]
ping @dpobel

File: Loader/ComboLoader.php [hunk 2]
```diff
@@ -162,4 +178,31 @@ private function findFileExtension($file)
...
+    }
+
+    private function renderTemplate($file)
+    {
+        preg_match('#/tpl/handlebars/(.*)_js#', $file, $matches);
+        $module = $matches[1];
+
+        $path = $this->getModuleSetting($module, 'path');
+
+        return $this->templating->render(
```
Comment (dpobel) [inline Loader/ComboLoader.php]: this is kind of duplicating https://github.com/ezsystems/PlatformUIBundle/blob/master/Controller/TemplateController.php#L42
Can't this be reused somehow ?

Comment (bdunogier) [inline Loader/ComboLoader.php]: I'm afraid it will have to be part of the cleanup follow-up...

**Comment by andrerom** [comment]
+1 :ship:

**Comment by bdunogier** [comment]
> +1 :ship:

love it :)

File: Loader/ComboLoader.php (hunk 2 above)
Comment (dpobel) [inline Loader/ComboLoader.php]: created https://jira.ez.no/browse/EZP-25174, you can put this feature in the :ship: :D

**Comment by bdunogier** [comment]
Merged. I have moved the issues forward and added the links.

**Pull request merged**